packages/contact-center/user-state/src/helper.ts

```diff
@@ -15,6 +16,16 @@ export const useUserState = ({idleCodes, agentId, cc}) => {
       setElapsedTime(prevTime => prevTime + 1);
     }, 1000);
 
+    cc.on(AGENT_STATE_CHANGE, (data) => {
```

Comment:
We will need to switch off this listener, and we will need to move the business logic to the callback function

Reply:
Coderabbit has the same comment below. Let's take care of this.

Reply:
Addressed